hephaistus/main-theme/distribution-system/chord/playground/nullist0/lib/src/test/kotlin/org/lia/dht/nullist0/chord/event/FindSuccessorByIdTest.kt

```diff
@@ -0,0 +1,166 @@
+package org.lia.dht.nullist0.chord.event
+
+import kotlinx.coroutines.test.runTest
+import org.lia.dht.nullist0.chord.ChordConfiguration
+import org.lia.dht.nullist0.chord.fake.TestChordIdentifierScope
+import org.lia.dht.nullist0.chord.fake.TestChordProtocol
+import org.lia.dht.nullist0.chord.fake.TestChordRepository
+import org.lia.dht.nullist0.chord.model.ChordNode
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class FindSuccessorByIdTest {
+    private val findSuccessorById = FindSuccessorById<UByte, UByte>(ID_TO_FIND)
+
+    @Test
+    fun testFindSuccessorById_equalNodeId() = runTest {
+        // given
+        val myNode = ChordNode.NetworkNode<UByte>("my").let { it.copy(successor = it) }
+        val scope = TestChordIdentifierScope(
+            nodeToIdMap = mapOf(myNode to ID_TO_FIND)
+        )
+        val configuration = ChordConfiguration(
+            node = myNode,
+            identifierScope = scope,
+            protocol = TestChordProtocol(scope, nodeToDataMap = mapOf(myNode to emptyMap())),
+            repository = TestChordRepository()
+        )
+
+        // when
+        val successor = findSuccessorById(configuration)
+
+        // then
+        assertEquals(myNode, successor)
+    }
+
+    @Test
+    fun testFindSuccessorById_largeNodeId() = runTest {
+        // given
+        val myNode = ChordNode.NetworkNode<UByte>("my").let { it.copy(successor = it) }
+        val scope = TestChordIdentifierScope(
+            nodeToIdMap = mapOf(myNode to LARGE_ID)
+        )
+        val configuration = ChordConfiguration(
+            node = myNode,
+            identifierScope = scope,
+            protocol = TestChordProtocol(scope, nodeToDataMap = mapOf(myNode to emptyMap())),
+            repository = TestChordRepository()
+        )
+
+        // when
+        val successor = findSuccessorById(configuration)
+
+        // then
+        assertEquals(myNode, successor)
+    }
+
+    @Test
+    fun testFindSuccessorById_smallNodeId() = runTest {
+        // given
+        val myNode = ChordNode.NetworkNode<UByte>("my").let { it.copy(successor = it) }
+        val scope = TestChordIdentifierScope(
+            nodeToIdMap = mapOf(myNode to SMALL_ID)
+        )
+        val configuration = ChordConfiguration(
+            node = myNode,
+            identifierScope = scope,
+            protocol = TestChordProtocol(scope, nodeToDataMap = mapOf(myNode to emptyMap())),
+            repository = TestChordRepository()
+        )
+
+        // when
+        val successor = findSuccessorById(configuration)
+
+        // then
+        assertEquals(myNode, successor)
+    }
+
+    @Test
+    fun testFindSuccessorById_twoNodes_with_otherNode_dataNode_myNode() = runTest {
```

Comment:
Q. It's inconsistent with upper test method name. Why do you prefer to mixuse camelCase and snake_case ? `testFindSuccessorById_twoNodesWithOtherDataNodeAndMyNode` is my understanding of your intention.